File: Leetcode/27. Remove Element/remove_element.py
class Solution:
    def removeElement(self, nums, val: int) -> int:
        index = 0
        for i in range(len(nums)):
            if nums[i] !=val:
                nums[index] = nums[i]
                index+=1
        return index


sol = Solution()
print(sol.removeElement([0,1,2,2,3,0,4,2],2))

File: Leetcode/27. Remove Element/remove_element.rs
struct Solution {}

impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut index =0;
        for i in 0..nums.len(){
            if nums[i]!=val{
                nums[index] = nums[i];
                index +=1;
            }
        }
        index as i32
    }
}
fn main() {
    let res = Solution::remove_element(&mut vec![0,1,2,2,3,0,4,2],2);
    println!("{:?}", res);
}
